--- expect-5.39/exp_command.c.orig	2003-07-31 20:21:42.000000000 +0200
+++ expect-5.39/exp_command.c	2003-08-07 22:11:03.000000000 +0200
@@ -801,7 +801,7 @@
 	    rfd = (int) rfdc;
 	}
 	if (mode & TCL_WRITABLE) {
-	    if (TCL_ERROR == Tcl_GetChannelHandle(channel, TCL_WRITABLE, (ClientData) &wfdc)) {
+	    if (TCL_ERROR == Tcl_GetChannelHandle(channel, TCL_WRITABLE, (ClientData *) &wfdc)) {
 		return TCL_ERROR;
 	    }
 	    wfd = (int) wfdc;
