diff -Nru expect-5.26/Makefile.in expect-5.26.new/Makefile.in
--- expect-5.26/Makefile.in	Tue Jun 16 15:53:36 1998
+++ expect-5.26.new/Makefile.in	Fri Sep 11 13:17:45 1998
@@ -410,8 +410,6 @@
 	$(INSTALL_DATA) $(srcdir)/expect_tcl.h $(includedir)
 	$(INSTALL_DATA) $(srcdir)/expect_comm.h $(includedir)
 # force installation of Tcl's private regexp definition - we simply have to
-# make it public in order for people to use Expect's C lib.
-	$(INSTALL_DATA) $(TCLHDIR)/tclRegexp.h $(includedir)
 # install Debugger's public include file (just in case it's not there)
 	$(INSTALL_DATA) $(srcdir)/Dbg.h $(includedir)
 # some people don't install Tcl, sigh
