--- expect5.31/fixcat.fixcat	Fri Jun  2 13:35:51 2000
+++ expect5.31/fixcat	Fri Jun  2 13:36:56 2000
@@ -8,11 +8,14 @@
 # If this file is sitting in an architecture-specific library directory,
 # then it serves as a marker that your /bin/cat buffers by default.
 
+# XXX Red Hat linux has unbuffered cat
+exit 0
+
 # test for when catting to/from files
-log_user 0
-spawn -open [open "|cat" "r+"]
-send "\r"
-expect "\r" {exit 1}
+#log_user 0
+#spawn -open [open "|cat" "r+"]
+#send "\r"
+#expect "\r" {exit 1}
 
 # test for when catting to real tty
 #log_user 0
