--- expect-5.44.1.15/tclconfig/tcl.m4.orig	2010-03-11 20:06:30.000000000 +0100
+++ expect-5.44.1.15/tclconfig/tcl.m4	2010-10-02 12:25:10.095738529 +0200
@@ -3149,7 +3149,7 @@
 	MAKE_STUB_LIB="\${STLIB_LD} -out:\[$]@ \$(PKG_STUB_OBJECTS)"
     else
 	MAKE_STATIC_LIB="\${STLIB_LD} \[$]@ \$(PKG_OBJECTS)"
-	MAKE_SHARED_LIB="\${SHLIB_LD} -o \[$]@ \$(PKG_OBJECTS) \${SHLIB_LD_LIBS}"
+	MAKE_SHARED_LIB="\${SHLIB_LD} -o \[$]@ \$(PKG_OBJECTS) \${SHLIB_LD_LIBS} -Wl,-soname,\[$]@"
 	MAKE_STUB_LIB="\${STLIB_LD} \[$]@ \$(PKG_STUB_OBJECTS)"
     fi
 
