diff -durN expect-5.39.orig/configure.in expect-5.39/configure.in
--- expect-5.39.orig/configure.in	2004-12-28 16:49:51.000000000 +0000
+++ expect-5.39/configure.in	2004-12-28 16:53:24.019681264 +0000
@@ -1132,7 +1132,7 @@
 fi
 
 EXP_BUILD_LIB_SPEC="-L`pwd` -lexpect${EXP_LIB_VERSION}${DBGX}"
-EXP_LIB_SPEC="-L\${INSTALL_ROOT}\${exec_prefix}/lib -lexpect${EXP_LIB_VERSION}${DBGX}"
+EXP_LIB_SPEC="-L\${INSTALL_ROOT}\${exec_prefix}/lib64 -lexpect${EXP_LIB_VERSION}${DBGX}"
 EXP_UNSHARED_LIB_FILE=libexpect${EXP_LIB_VERSION}${DBGX}.a
 
 # The TCL_SHARED_LIB_SUFFIX macro below relies on the DBGX macro,
